src/printer.js

```diff
@@ -1131,6 +1131,8 @@ function printExpression(path, options, print) {
           ])
         );
       }
+      const isAssociative = !!node.items[0].key;
+      const shouldBreak = isAssociative && node.loc.source.includes("\n");
```

Comment:
Is this what JS does? Seems a little strange to me, but I'm fine w/ it if so

Reply:
Do you mean the check for `\n`? JS is calling `hasNewlineInRange`, which might be safer ([see here](https://github.com/prettier/prettier/blob/a938076ae5a43e2eea1fb7a0a113e7e038076f55/src/language-js/printer-estree.js#L1111))

Reply:
@czosel let's use `hasNewlineInRange` :+1: ,maybe we should create issue about move this to shared utils in prettier

Reply:
Ok, I'll refactor this PR based on `hasNewlineInRange` and open a PR in prettier core that adds it. Only downside would be that we'd have to point to a specific (unreleased) prettier core version in our `package.json` again :stuck_out_tongue:

Reply:
@czosel we can add own `hasNewlineInRange` util right now and after upgrade remove this code from utils (just left TODO about this)